### DISCUSSION TITLE
Ability to clean the schema

### DISCUSSION THREAD

**Discussion by dxnxk-0** [rust-db/refinery#268] [General]
Is there any way to clean the schema conveniently without the need to drop/create the database ? Thanks for your help.

**Comment by jxs** [comment]
Hi, can you elaborate on what cleaning the schema means then?
Thanks!

**Comment by dxnxk-0** [comment]
Basically the ability to "clean" the schema, hence dropping all the tables: https://flywaydb.org/documentation/command/clean

Very convenient in dev. Alternatively, diesel provides the "redo" migration command which allow to rollback and run the last transaction.

It avoids the pain of `dropdb && createdb && migrate` at each schema iteration.

**Comment by jxs** [comment]
Interesting! Yeah this also makes sense, would you want to go ahead and try to implement it? I can guide you through wherever you need

**Reply by ririsoft** [reply]
I am interested in this topic too and open to contribute.

My use case is that during development or testing I want to re-run my migration from V1.

I usually `DROP TABLE IF EXISTS refinery_schema_history` in my migration tests. I also make sure that all my migration scripts can be re-executed  (using `DROP TABLE IF EXISTS` before creating a table for instance).

Having a `runner().redo(target).run(conn)` helper for this would be helpful since it does not require refinery users to understand its internals. However this is not magic and it is still up to the developers to make sure their migrations can be re-run from scratch.

What is your view on what refinery should (or should not) do in such use cases ?

**Reply by jxs** [reply]
Hi, I am really not found of undo or redo migrations, see https://github.com/rust-db/refinery/issues/121 but a clean command I think may make sense as we can call `runner.run()` and `run_async()` after again without any hassle I'd say, would you like to go further and try to implement it?
Thanks!
